### PR TITLE
Add validate to builder unit-test

### PR DESCRIPTION
**Pull request by Bento007**
- make unit-test fixtures reusable.
- Test for correct directory structor in `validate`.
- add validate to builder unit test

This issue completes #195, an partially complete #11

### CODE REVIEW

File: tools/cell_census_builder/tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    uns["batch_condition"] = np.array(["a", "b"], dtype="object")
+
+    # Need to carefully set the corpora schema versions in order for tests to pass.
+    uns["schema_version"] = "3.0.0"  # type: ignore
+
+    return anndata.AnnData(X=X, obs=obs, var=var, obsm=obsm, uns=uns)
+
+
+@pytest.fixture
+def assets_path(tmp_path: pathlib.Path) -> str:
```
Comment (ebezzi) [inline tools/cell_census_builder/tests/conftest.py:79]: Slightly confused - shouldn't there be a fixture for `tmp_path` as well?

File: tools/cell_census_builder/tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    return [
+        Dataset(dataset_id="first_id", corpora_asset_h5ad_uri="mock", dataset_h5ad_path=first_h5ad_path),
+        Dataset(dataset_id="second_id", corpora_asset_h5ad_uri="mock", dataset_h5ad_path=second_h5ad_path),
+    ]
+
+
+@pytest.fixture()
+def setup(monkeypatch: MonkeyPatch) -> None:
+    process_initializer()
+    monkeypatch.setitem(
```
Comment (ebezzi) [inline tools/cell_census_builder/tests/conftest.py:112]: 👍

File: tools/cell_census_builder/tests/conftest.py (hunk 1 above)
Comment (Bento007) [reply tools/cell_census_builder/tests/conftest.py:79]: this is pytest magic. I don't love it but it's pretty convenient. https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html

Comment (bkmartinjr) [reply tools/cell_census_builder/tests/conftest.py:79]: it is very well known to pytest users.  There are quite a few built-ins and they are part of the pytest "charm" :-)

Comment (Bento007) [reply tools/cell_census_builder/tests/conftest.py:79]: It's growing on on me.

**Pull request merged**